src/Model/Profile.php

```diff
@@ -273,8 +263,17 @@ private static function sidebar(App $a, array $profile, $block = 0, $show_connec
 		$o = '';
 		$location = false;
 
-		// This function can also use contact information in $profile
-		$is_contact = !empty($profile['cid']);
+		// This function can also use contact information in $profile, but the 'cid'
+		// value is going to be coming from 'owner-view', which means it's the wrong
+		// contact ID for the user viewing this page. Use 'nurl' to look up the
+		// correct contact table entry for the logged-in user.
+		$is_contact = !empty($profile['nurl']);
+		$contact = [];
+
+		if ($is_contact) {
```

Comment:
It's a much better approach, thanks. I only want to suggest a small change. To avoid that this code is executed when you are visiting your own profile, you could check if `$profile['uid']` differs from `local_user()` (and of course if `local_user()` is non empty at all).

Reply:
Done! :)